### PR TITLE
boards: rp2040: Add user gpio driver

### PR DESCRIPTION
**Pull request by microhobby**
Add user space gpio driver supporting gpout, gpin, gpint

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

## Summary

Based on the user space driver from ESP32

## Impact

GPIO access support for user space applications

## Testing

Tested using `nsh` and `gpio` example from apps

### CODE REVIEW

**Comment by microhobby** [comment]
Hmm, I see that my patch is not passing on `check` validation. Let me solve this ...

**Comment by microhobby** [comment]
OK, check passed (/me remember to run `checkpatch.sh` 😅)

File: boards/arm/rp2040/raspberrypi-pico/src/rp2040_gpio.c [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+
+/* This array maps the GPIO pins used as OUTPUT */
+
+static const uint32_t g_gpiooutputs[BOARD_NGPIOOUT] =
+{
+  GPIO_OUT1,
+  GPIO_OUT2,
+  GPIO_OUT3,
+  GPIO_OUT4,
+  GPIO_OUT5
```
Comment (acassis) [inline boards/arm/rp2040/raspberrypi-pico/src/rp2040_gpio.c:115]: @microhobby you declared BOARD_NGPIOOUT = 1 but included 5 items inside the array, please remove the remaining items.

File: boards/arm/rp2040/raspberrypi-pico/src/rp2040_gpio.c [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+};
+
+/* This array maps the GPIO pins used as INTERRUPT INPUTS */
+
+static const uint32_t g_gpioinputs[BOARD_NGPIOIN] =
+{
+  GPIO_IN1,
+  GPIO_IN2,
+  GPIO_IN3,
+  GPIO_IN4
```
Comment (acassis) [inline boards/arm/rp2040/raspberrypi-pico/src/rp2040_gpio.c:137]: ditto

File: boards/arm/rp2040/raspberrypi-pico/src/rp2040_gpio.c (hunk 1 above)
Comment (microhobby) [reply boards/arm/rp2040/raspberrypi-pico/src/rp2040_gpio.c:115]: oops, this come from the previous application. BTW there is a way to add it by configuration (let say like device trees from Linux?)

File: boards/arm/rp2040/raspberrypi-pico/src/rp2040_gpio.c (hunk 2 above)
Comment (microhobby) [reply boards/arm/rp2040/raspberrypi-pico/src/rp2040_gpio.c]: Thanks for the review, fixed

File: boards/arm/rp2040/raspberrypi-pico/src/rp2040_gpio.c (hunk 1 above)
Comment (acassis) [reply boards/arm/rp2040/raspberrypi-pico/src/rp2040_gpio.c]: There is not a config to do it. But in you can declare the variable without its size: g_gpiooutputs[] and use sizeof() to discover its size. There are some boards already doing it.

**Pull request merged**